src/app/(layout-with-navigation)/personas/_helpers/getPersonaFoldedSteps.ts

```diff
@@ -0,0 +1,62 @@
+import getQuestionsOfMosaic from '@/publicodes-state/helpers/getQuestionsOfMosaic'
+import { safeGetSituation } from '@/publicodes-state/helpers/safeGetSituation'
+import {
+  DottedName,
+  Engine,
+  NGCEvaluatedNode,
+  NGCRuleNode,
+} from '@/publicodes-state/types'
+import { Situation } from '@/types/simulation'
+import { fixSituationWithPartialMosaic } from './fixSituationWithPartialMosaic'
+
+export const getPersonaFoldedSteps = (
+  situation: Situation,
+  everyMosaic: DottedName[],
+  everyMosaicChildren: DottedName[],
+  everyQuestions: DottedName[],
+  everyRules: DottedName[],
+  pristineEngine: Engine,
+  safeGetRule: (rule: DottedName) => NGCRuleNode | null,
+  safeEvaluate: (rule: DottedName) => NGCEvaluatedNode | null
+) => {
+  const personaSituation = fixSituationWithPartialMosaic(
+    situation,
+    everyMosaic,
+    everyMosaicChildren,
+    safeGetRule,
+    safeEvaluate
+  )
+
+  const safeSituation = safeGetSituation({
+    situation: personaSituation,
+    everyRules,
+  })
+
+  pristineEngine.setSituation(safeSituation)
+
+  const missingVariables = Object.keys(
+    pristineEngine.evaluate('bilan')?.missingVariables || {}
+  ).filter((missingVariable) => everyQuestions.includes(missingVariable))
+
+  const personaFoldedSteps = missingVariables
+
+  everyMosaic.forEach((mosaic) => {
+    const expectedMosaicGroup = getQuestionsOfMosaic({
+      dottedName: mosaic,
+      everyMosaicChildren,
+    })
+    let isMosaicInSituation = false
+    expectedMosaicGroup.forEach((dottedName) => {
+      const index = personaFoldedSteps.indexOf(dottedName)
```

Comment:
Globalement, une bonne règle à suivre pour déterminer si une constante est requise : 
- est-ce que le code qu'elle contient est utilisé plus d'une fois
- est-ce qu'elle permet de gagner en clarté / libisilité

Ici je suis pas convaincu du deuxième point

Reply:
en fait je verrais bien :
```suggestion
      const isDottedNameInPersonaFoldedSteps = personaFoldedSteps.indexOf(dottedName) > -1
      if (isDottedNameInPersonaFoldedSteps) {
```

Reply:
Elle est bien utilisée pour le splice apres

Reply:
Ah oui, mon erreur 🤓